File: app/src/main/java/com/lmyby/ankihelper/domain/OnlockReceiver.java
package com.lmyby.ankihelper.domain;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

import com.lmyby.ankihelper.data.Settings;

public class OnlockReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        Intent newIntent = new Intent(context, CBWatcherService.class);
        Settings settings = Settings.getInstance(context);
        if(settings.getMoniteClipboardQ()) {
            Toast.makeText(context, "debug", Toast.LENGTH_SHORT).show();
            context.startService(newIntent);
        }
    }
}

File: app/src/main/java/com/lmyby/ankiquicker/domain/OnlockReceiver.kt
package com.lmyby.ankiquicker.domain

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.widget.Toast
import com.lmyby.ankiquicker.data.Settings

/**
 * BroadcastReceiver for screen lock/unlock events
 * Converted to Kotlin as part of Phase 3 domain logic migration
 */
class OnlockReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val newIntent = Intent(context, CBWatcherService::class.java)
        val settings = Settings.getInstance(context)
        if (settings.getMoniteClipboardQ()) {
            Toast.makeText(context, "debug", Toast.LENGTH_SHORT).show()
            context.startService(newIntent)
        }
    }
}
